### PR TITLE
Added ability to change frame colour

### LINKED ISSUE #262: Suggestion: adjust frame background color
Hi,

I have a suggestion, not sure if it's even technically possible.

The problem:
When you use HoverZoom+ on a mostly black website, and it's not loading an image right away, then it 'flashes' the white frame background for a short period of time (<1s) before loading the image. The effect is less in-your-face on lighter sites.

The solution:
This could be fixed by either
- calculating the average color value of the page Chrome is currently displaying and using that to set the frame background color accordingly
or by
- allowing the user to manually set the frame background color

I'm using Windows 10, latest Chrome and HoverZoom+ version.

### PR DESCRIPTION
**Pull request by ajlomagno**
Fixes #262

This pr adds the ability for the user to choose their own colour for the hoverzoom frame. So far the new option in the options screen is only in english.

Choosing a new colour:
part1

The new colour in action:
part2

### CODE REVIEW

File: js/common.js [hunk 1]
```diff
@@ -34,6 +34,7 @@ function loadOptions() {
     options.ambilightEnabled = options.hasOwnProperty('ambilightEnabled') ? options.ambilightEnabled : false;
     options.disabledPlugins = options.hasOwnProperty('disabledPlugins') ? options.disabledPlugins : [];
     options.centerImages = options.hasOwnProperty('centerImages') ? options.centerImages : false;
+    options.frameBackgroundColour = options.hasOwnProperty('frameBackgroundColour') ? options.frameBackgroundColour : "#ffffff";
```
Comment (extesy) [inline js/common.js]: Can you please follow American spelling conventions (colour -> color)?

File: js/hoverzoom.js [hunk 2]
```diff
@@ -124,7 +123,7 @@ var hoverZoom = {
             if (position === undefined || position.top === undefined || position.left === undefined) {
                 position = {top:mousePos.top, left:mousePos.left};
             }
-
+
```
Comment (extesy) [inline js/hoverzoom.js]: please remove whitespace on this line

File: js/common.js (hunk 1 above)
Comment (extesy) [reply js/common.js]: And also make this change in all other files as well.

File: js/options.js [hunk 3]
```diff
@@ -143,8 +144,16 @@ function restoreOptions() {
     $('#txtFadeDuration').val((options.fadeDuration || 0) / 1000);
     $('#chkAmbilightEnabled')[0].checked = options.ambilightEnabled;
     $('#chkCenterImages')[0].checked = options.centerImages;
+    $('#pickerFrameBackgroundColour').val(options.frameBackgroundColour);
     $('#selectCaptionLocation').val(options.captionLocation);
 
+    if(options.frameBackgroundColour == ""){
```
Comment (extesy) [inline js/options.js]: please add spaces before `(` and after `)`

File: js/options.js [hunk 4]
```diff
@@ -143,8 +144,16 @@ function restoreOptions() {
...
+        initColourPicker('#ffffff');
+    }else{
```
Comment (extesy) [inline js/options.js]: please add spaces around `else`

File: js/options.js [hunk 5]
```diff
@@ -143,8 +144,16 @@ function restoreOptions() {
...
+        initColourPicker(options.frameBackgroundColour);
+    }
+
+
```
Comment (extesy) [inline js/options.js]: remove extra empty line

File: js/options.js [hunk 6]
```diff
@@ -287,6 +296,20 @@ function populatePluginsTable() {
...
+        color: colour,
+        preferredFormat: "hex",
+        chooseText: "Select",
+        cancelText: "Cancel",
+        change:  function(color){
+            $('#pickerFrameBackgroundColour').attr('value', color.toHexString());
+        }
+    })
+
+    //colorPicker.attr('value', colorPicker.spectrum('get'));
```
Comment (extesy) [inline js/options.js]: remove this commented out line and previous empty line

File: js/options.js [hunk 7]
```diff
@@ -287,6 +296,20 @@ function populatePluginsTable() {
     Gumby.initialize('checkbox');
 }
 
+function initColourPicker(colour){
+    var colorPicker = $('#pickerFrameBackgroundColour').spectrum({
+        color: colour,
+        preferredFormat: "hex",
+        chooseText: "Select",
```
Comment (extesy) [inline js/options.js]: this needs to be localized (and `cancelText` as well)

File: js/options.js [hunk 8]
```diff
@@ -287,6 +296,20 @@ function populatePluginsTable() {
...
+        cancelText: "Cancel",
+        change:  function(color){
```
Comment (extesy) [inline js/options.js]: remove extra space after `:` and add a space before `{`

**Comment by ajlomagno** [comment]
No problem! I'll take a look at what you wrote and fix up the pr. I originally intended to spell colour the American way but I guess I subconsciously wrote it the way I'm used to instead.

**Comment by ajlomagno** [comment]
I fixed the issues mentioned above and added a simple color contrast implementation so the caption text color will change depending on the chosen frame color to keep the caption readable.

**Pull request merged**